packages/react/src/table/TableHeaderSortable.tsx

```diff
@@ -0,0 +1,77 @@
+import { Box, Flex } from '../box';
+import { BaseButton } from '../button';
+import { boxPalette } from '../core';
+import { ArrowDownIcon, ArrowUpIcon } from '../icon';
+import { TableHeaderProps } from './TableHeader';
+
+export type TableSortDirection = 'ASC' | 'DESC' | undefined;
+
+type TableHeaderSortableProps = TableHeaderProps & {
+	sort: TableSortDirection;
+	onClick?: () => void;
+};
+
+/** Used in place of TableHeader when users can click a column header to
+ * sort the table. */
+export const TableHeaderSortable = ({
+	children,
```

Comment:
There are some props here like `display`, `scope` etc that are not being used. I wonder if it's best to try and re-use the `TableHeader` here.

Reply:
I've narrow the types using `Pick<>`